public/components/trace_analytics/components/traces/span_detail_flyout.tsx

```diff
@@ -74,6 +74,31 @@ const getSpanValue = (span: object, mode: TraceAnalyticsMode, field: SpanField)
   return get(span, fieldKey);
 };
 
+// Flatten nested objects with dot notation
+const flattenObject = (
```

Comment:
Can we add some unit tests for this?

Reply:
Added the following unit test:
<img width="858" alt="UnitTest" src="https://github.com/user-attachments/assets/68f319d0-25b3-4365-868a-03985b7c0aa4" />